lib/matplotlib/dates.py

```diff
@@ -1254,6 +1241,12 @@ def __init__(self, bymonthday=None, interval=1, tz=None):
 
         Default is to tick every day of the month: ``bymonthday=range(1,32)``
         """
+        if interval < 1:
+            raise ValueError("The interval parameter must be an integer "
```

Comment:
That makes sense. I thought of both cases. On the one hand, I feel since the parameter is being addressed anyways, it is worthwhile to flat out say both limitations. But on the other, `if interval <1 or not isinstance(interval, int)` started looking a little full. However, I think it is still the better option. I will consolidate it to one check/one exception.